src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,331 @@
+#include "locomotor.h"
+/**
+ * @file LocomotorStore
+ *
+ * @author Duncans_pumpkin
+ *
+ * @brief LocomotorStore
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#ifndef GAME_DLL
+LocomotorStore *g_theLocomotorStore = nullptr;
+#endif
+
+// zh: 0x004B8260 wb: 0x0074E62F
+LocomotorStore::~LocomotorStore()
+{
+    for (auto &iter : m_locomotorTemplates) {
+        iter.second->Delete_Instance();
+    }
+    m_locomotorTemplates.clear();
+}
+
+// zh: 0x004B84B0 wb: 0x0074E7B0
+void LocomotorStore::Reset()
+{
+    for (auto it = m_locomotorTemplates.begin(); it != m_locomotorTemplates.end();) {
+        it->second = static_cast<LocomotorTemplate *>(it->second->Delete_Overrides());
+        if (it->second == nullptr) {
+            m_locomotorTemplates.erase(it);
```

Comment:
This is erasing iterator, and then using it in next iteration. I am not sure if this is proper for map right now. It may not be.